src/best_offset_prefetcher.h

```diff
@@ -0,0 +1,121 @@
+/** $lic$
+ *
+* This file is part of zsim.
+ *
+ * zsim is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 2.
+ *
+ * If you use this software in your research, we request that you reference
+ * the zsim paper ("ZSim: Fast and Accurate Microarchitectural Simulation of
+ * Thousand-Core Systems", Sanchez and Kozyrakis, ISCA-40, June 2013) as the
+ * source of the simulator in any publications that use this software, and that
+ * you send us a citation of your work.
+ *
+ * zsim is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+//#define TEST
+//#define SIMPLE_PC
+
+#ifndef BEST_OFFSET_PREFETCHER_H_
+#define BEST_OFFSET_PREFETCHER_H_
+
+#ifndef TEST 
+#include "cache_prefetcher.h"
+#include "filter_cache.h"
+#include "g_std/g_string.h"
+#include "memory_hierarchy.h"
+#include "stats.h" 
+#include "hash.h"
+// hopefully don't need to include this for tests in the .cpp
+#else 
+#include <map>
+#include <vector>
+#include <iostream>
+#include <cmath> 
+#include <cmath> 
+#endif
+
+// num run throughs to make before round over
+const uint64_t RND_MAX = 100;
+//const uint64_t RND_MAX = 50;
```

Comment:
remove multiple constants. If you believe these should be configurable, add a new parameter that can be defined via the cfg file.